FirebaseSessions/Sources/EventFireLogger.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@_implementationOnly import GoogleDataTransport
+
+protocol EventFireLoggerProtocol {
+  func logEvent(event: SessionStartEvent, completion: @escaping (Result<Void, Error>) -> Void)
+}
+
+class EventFireLogger: EventFireLoggerProtocol {
```

Comment:
I worry that EventDispatcher would be confusing because on iOS Dispatch is used to convey threading, where this class is responsible for logging to FireLog. The method on GDT is also called `logGDTEvent`. I understand that logging also pertains to printing to console logs, but in this case I do think EventFireLogger is very clear.

Maybe `EventGDTLogger`?